packages/grafana-schema/src/common/data_frame.cue

```diff
@@ -0,0 +1,217 @@
+		// An explict path to the field in the datasource.  When the frame meta includes a path,
+		// This will default to `${frame.meta.path}/${field.name}
+		// When defined, this value can be used as an identifier within the datasource scope, and
+		// may be used to update the results
+		path?: string
+		// True if data source can write a value to the path.  Auth/authz are supported separately
+		writeable?: bool
+		// True if data source field supports ad-hoc filters
+		filterable?: bool
+		// Numeric Options
+		unit?: string
+		// Significant digits (for display) TODO this should be a separate type
+		decimals?: int64
+		// TODO | null
+		min?: int64
+		// TODO | null
+		max?: int64
+		// Interval indicates the expected regular step between values in the series.
+		// When an interval exists, consumers can identify "missing" values when the expected value is not present.
+		// The grafana timeseries visualization will render disconnected values when missing values are found it the time field.
+		// The interval uses the same units as the values.  For time.Time, this is defined in milliseconds.
+		// TODO | null
+		interval?: int64
+		// Convert input values into a display string
+		mappings?: [...ValueMapping]
+		// Map numeric values to states
+		thresholds?: [...ThresholdsConfig]
+} @cuetsy(kind="interface")
+
+// TODO this doesn't work
+//DecimalCount?: int64 | null @cuetsy(kind="type")
+
+FieldState: {
+		// An appropriate name for the field (does not include frame info) TODO | null
+		displayName?: string
+		//Cache of reduced values
+		calcs?: FieldCalcs
+		// The numeric range for values in this field.  This value will respect the min/max
+		// set in field config, or when set to `auto` this will have the min/max for all data
+		// in the response
+		range?: NumericRange
+		// Appropriate values for templating
+		scopedVars?: ScopedVars
+		// Series index is index for this field in a larger data set that can span multiple DataFrames
+		// Useful for assigning color to series by looking up a color in a palette using this index
+		seriesIndex?: int64
+		// Location of this field within the context frames results
+		// @internal -- we will try to make this unnecessary
+		origin?: DataFrameFieldIndex
+		// Boolean value is true if field is in a larger data set with multiple frames.
+		// This is only related to the cached displayName property above.
+		multipleFrames?: bool
+		// Boolean value is true if a null filling threshold has been applied
+		// against the frame of the field. This is used to avoid cases in which
+		// this would applied more than one time.
+		nullThresholdApplied?: bool
+} @cuetsy(kind="interface")
+
+// TODO Docs
+FieldCalcs: {...}
+
+// TODO docs
+NumericRange: {
+		min?: int64 // TODO | null
+		max?: int64 // TODO | null
+		delta: int64
+} @cuetsy(kind="interface")
+
+DataFrameFieldIndex: {
+		frameIndex: int64
+		fieldIndex: int64
+} @cuetsy(kind="interface")
+
+// TODO Duplicate declaration
```

Comment:
@sdboyer I've tried moving these duplicate types into `common_dataquery.cue` but they still end up in `common.gen.ts`, any idea how to make them available in cue but do not include in generated files, since these types are already present in dashboard types?

Reply:
i thiiiink if you just omit the `@cuetsy` attributes on them, it should do what you want?

Reply:
But then they will be just generated inline and not available for use in other cue files?

Reply:
oh. yeah. 🤦🏻 

will have to think on these a bit